src/Featurizers/AnalyticalRollingWindowFeaturizer.h

```diff
@@ -0,0 +1,305 @@
+#include "../Featurizer.h"
+#include "../Traits.h"
+
+#include <vector>
+
+namespace Microsoft {
+namespace Featurizer {
+namespace Featurizers {
+
+static constexpr char const * const         AnalyticalRollingWindowEstimatorName("AnalyticalRollingWindowEstimator");
+
+/////////////////////////////////////////////////////////////////////////
+///  \fn            AnalyticalRollingWindowCalculation
+///  \brief         Which type of calculation will be performed on the rolling window.
+///                 Currently only one of calculation is supported. This will be expanded upon in the future,
+///                 so we are peparing by making this an enum from the get go.
+///
+enum class AnalyticalRollingWindowCalculation : std::uint8_t {
+    Mean = 1
+};
+
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         AnalyticalRollingWindowTransformer
+///  \brief         RollingWindow class that is used for all analytical window computations.
+///
+template <
+    typename InputT,
+    size_t MaxNumTrainingItemsV=std::numeric_limits<size_t>::max()
+>
+class AnalyticalRollingWindowTransformer:
+    public Components::InferenceOnlyTransformerImpl<InputT, std::vector<double>> {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using BaseType = Components::InferenceOnlyTransformerImpl<InputT, std::vector<double>>;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    AnalyticalRollingWindowTransformer(std::uint32_t windowSize, AnalyticalRollingWindowCalculation windowCalculation, std::uint32_t horizon, std::uint32_t minWindowCount = 1) ;
+    AnalyticalRollingWindowTransformer(Archive &ar);
+    ~AnalyticalRollingWindowTransformer(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(AnalyticalRollingWindowTransformer);
+
+    bool operator==(AnalyticalRollingWindowTransformer const &other) const;
+    bool operator!=(AnalyticalRollingWindowTransformer const &other) const;
+    void save(Archive & ar) const override;
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Members
+    // |
+    // ----------------------------------------------------------------------
+    // _windowSizes and _minWindowCount are int64 so that we can have the full positive range of the uint32 it comes
+    // in as but can still support negative numbers.
+    Components::CircularBuffer<InputT>              _buffer;
+    const std::uint32_t                             _windowSize;
+    const std::uint32_t                             _horizon;
+    const std::uint32_t                             _minWindowSize;

```

Comment:
>const std::uint32_t                             _minWindowSize; [](start = 4, length = 63)

Nit: I'd move this closer to the declaration of `_windowSize` since they are related.

Reply:
I put it at the bottom since its optional in the constructor and thats is semi how I had them organized here as well. I will move it, but what is generally considered good practice for stuff like that?